src/main/java/org/icgc/argo/workflow_management/service/WorkflowStatusMonitor.java

```diff
@@ -0,0 +1,148 @@
+package org.icgc.argo.workflow_management.service;
+
+import static java.lang.String.format;
+import static java.time.OffsetDateTime.now;
+import static java.time.OffsetDateTime.parse;
+import static org.icgc.argo.workflow_management.util.JsonUtils.toJsonString;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import java.io.IOError;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.icgc.argo.workflow_management.service.model.Metadata;
+import org.icgc.argo.workflow_management.service.model.Workflow;
+import org.icgc.argo.workflow_management.service.model.WorkflowErrorEvent;
+import org.icgc.argo.workflow_management.service.properties.NextflowProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Slf4j
+@Service
+public class WorkflowStatusMonitor implements Runnable {
```

Comment:
A couple of points:

1. Since you have made your service class Runnable, and by default spring uses Singletons in its IoC container, that means there will only be one instance of the calling service and therefore a single invocation of its constructor which is what initializes this service thus creating the thread. As far as I can see unless I'm wrong, only a single new thread will ever be created. 

2. The only time WorkflowStatusMonitor is used, is through the use of the `new` keyword with the dependencies (NextflowProperties) being passed directly. There is no inversion of control. Since there is no DI happening I do not feel this needs to be a Service.